src/SonarQube.TeamBuild.Integration/CoverageReportProcessorBase.cs

```diff
@@ -117,6 +124,24 @@ private static void InsertCoverageAnalysisResults(string sonarOutputDir, string
             }
         }
 
+        /// <summary>
+        /// Insert code coverage results information into each projectinfo file
```

Comment:
This information just needs to be at global level, doesn't it? Since this code is only run in the _End_ step, it should be possible just to keep it in memory without writing it disk (in fact, if it's the same path for every project it shouldn't be necessary to write it out for each project).
The _AnalysisConfig_ class has a property bag for _LocalSettings_ that is a possible location for storing the data so it is passed to the properties writer.